src/components/design-library-list/util.js

```diff
@@ -127,34 +133,62 @@ export const addBackgroundScheme = (
 	}
 
 	if ( ! enableBackground ) {
-		return block
+		return blocks
 	}
 
-	const customAttributes = block.attributes.customAttributes
+	return blocks.map( block => {
+		const customAttributes = block.attributes.customAttributes
 
-	if ( ! customAttributes?.length ) {
-		block = addBackground( block )
-		return block
-	}
+		if ( ! customAttributes?.length ) {
+			block = addBackground( block )
+			return block
+		}
 
-	getTargetBlock( block.innerBlocks )
+		getTargetBlock( block.innerBlocks )
 
-	return block
+		return block
+	} )
 }
 
-export const replacePlaceholders = ( block, category ) => {
-	const key = block.attributes.text
-	const defaultValues = DEFAULT_CONTENT[ category ]
-	if ( key && defaultValues ) {
-		const newValue = defaultValues[ key ]
-		block.attributes.text = newValue
+export const adjustPatternSpacing = ( attributes, category, spacingSize, isDesignLibraryDevMode ) => {
+	// Note: The styles for block margin/padding will only be generated by the BlockCssGenerator,
+	// which updates the generatedCss attribute. Therefore, updating the block margin/padding
+	// attributes here will have no effect on the preview. Instead, we add additional CSS directly
+	// to the generatedCss attribute for preview purposes. This generatedCss will be regenerated
+	// by the BlockCssGenerator once the pattern is actually added to the editor.
+	let additionalCss = ''
+	if ( ! isDesignLibraryDevMode && category !== 'Header' ) {
+		if ( attributes.hasBackground ) {
+			additionalCss += 'margin-bottom: 0px !important;'
+		} else {
+			additionalCss += `margin-top: ${ spacingSize } !important;`
+			additionalCss += `margin-bottom: ${ spacingSize } !important;`
+		}
+
+		const blockLayouts = select( 'stackable/global-spacing-and-borders' ).getBlockLayouts()
+		if ( attributes.hasBackground && typeof blockLayouts === 'object' && ! blockLayouts[ 'block-background-padding' ] ) {
+			additionalCss += `padding-top: ${ spacingSize } !important;`
+			additionalCss += `padding-right: ${ spacingSize } !important;`
+			additionalCss += `padding-bottom: ${ spacingSize } !important;`
+			additionalCss += `padding-left: ${ spacingSize } !important;`
+		}
 	}
 
-	block.innerBlocks.forEach( innerBlock => {
-		replacePlaceholders( innerBlock, category )
-	} )
+	if ( additionalCss && attributes.uniqueId ) {
+		attributes.generatedCss += `.stk-${ attributes.uniqueId } {${ additionalCss }}`
+	}
+}
+
+export const replacePlaceholders = ( designContent, designCategory, isDesignLibraryDevMode ) => {
+	const defaultValues = DEFAULT_CONTENT[ designCategory ]
```

Comment:
Saw that we're using the category name here instead of a slug-type identifier. This might fail if we introduce some translations in the future. Change the keys here into slugs.